### PR TITLE
spanner-jdbc-cleanup - Step 06 - Add integration tests for JDBC driver

### PR DESCRIPTION
**Pull request by olavloite**
Adds integration tests for the JDBC driver. The integration tests also include an injector to abort transactions on demand in order to test transaction retry when aborted.

The integration tests use the same configuration class as the normal Spanner integration tests, i.e. they are read from `GceTestEnvConfig`. If no specific test environment variables have been set, the integration tests will use the default credentials and default project of the environment.

### CODE REVIEW

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/ITAbstractSpannerTest.java [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+  @BeforeClass
+  public static void setup() throws IOException, InterruptedException, ExecutionException {
+    createSpanner();
+    createInstance();
+    createDatabase();
+  }
+
+  private static void createSpanner() throws IOException {
+    // generate a unique instance id for this test run
+    Random rnd = new Random();
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/ITAbstractSpannerTest.java]: can we use the `IntegrationTestEnv` that already exists in the Spanner client lib tests?

Comment (olavloite) [reply google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/ITAbstractSpannerTest.java]: Yes, I think that is doable. I'll look into it.

**Comment by kolea2** [comment]
Separately, can you please break this PR up into smaller ones?

**Comment by olavloite** [comment]
Closing this in favor of smaller PR's using the standard Spanner `IntegrationTestEnv`.

**Pull request closed without merging**